### ISSUE TITLE
support for pydantic `extra=` model config

### ISSUE THREAD

**Issue by OlegZv** [astral-sh/ty#3503]
### Summary

Hello! I started getting `error[unknown-argument]` when running a type check on a Pydantic Class with a field that has an alias - starting with 0.0.33. In `ty==0.0.32` all checks pass. Here's a small demo:
```python
# demo.py
from pydantic import BaseModel, Field


class MyModel(BaseModel):
    name: str | None = None
    # changing alias= to validation_alias= or serialization_alias= makes the error go away
    demo_param: list[int] | None = Field(default=None, alias="DEMOParam")


mod = MyModel(demo_param=[])
```

```bash
# setup
uv init
uv add pydantic==2.13.4 ty==0.0.33
uv run ty check demo.py
```

I checked every version starting with 0.0.33 to 0.0.38 - all generate the error. Any help or guidance? 
Tried checking `__annotations__` to see if that's getting screwed up - `{'name': str | None, 'demo_param': list[int] | None}`. But I'm guessing it's something to do with how the resolver treats keyword `alias=`? Since changing the `alias=` to `serialization_alias` or `validation_alias` fixes it.

Thank you!

EDIT: Thought it was related to #1438, but This example does not use `populate_by_name`

### Version

0.0.33

**Comment by carljm** [comment]
Given the exact code provided, I think `MyModel(demo_param=[])` is actually a subtle bug? Because the [default model config](https://pydantic.dev/docs/validation/latest/concepts/alias/#alias-configuration) has `validate_by_alias=True` and `validate_by_name=False`, which means that `MyModel(demo_param=[])` does not populate the `demo_param` field; only `MyModel(DEMOParam=[])` would do that. `MyModel(demo_param=[])` uses the default for the `demo_param` field, and the constructor call only passes silently because `extra='ignore'` is [also the default behavior](https://pydantic.dev/docs/validation/latest/api/pydantic/config/#pydantic.config.ConfigDict.extra).

I think our behavior here is correct according to the `dataclass_transform` spec (and all other type checkers that don't special-case pydantic, including mypy without the pydantic plugin) agree with ty's error here. But if we had special-cased pydantic support, we should I guess let this constructor call pass due to the `extra='ignore'` default behavior.

**Comment by carljm** [comment]
Oh it looks like `validate_by_*` are already covered by #1438. But I don't think we have an issue for supporting `extra=`; we can use this issue for that.

**Comment by OlegZv** [comment]
Thank you for the quick and on-point response @carljm! In my actual code I do define `validate_by_name=True` as a config, but thought it was not relevant since the error would happen with and without it - explains why my code is working :) Thanks again!

**Comment by sharkdp** [comment]
Both `validate_by_{name,alias}` and `extra` are now supported. The original snippet doesn't lead to ty errors anymore, with and without `validate_by_name`, as expected.
